From 632eb9067ed2e606fd920902d0d2051134ce4e35 Mon Sep 17 00:00:00 2001
From: Pascal Quantin <pascal.quantin@gmail.com>
Date: Sat, 17 Oct 2015 17:12:46 +0200
Subject: [PATCH 62/90] UMTS FP: fix an out of bounds access

Bug: 11602
Change-Id: I636a5494a0eda5895e856e80424be29f01c758bf
Reviewed-on: https://code.wireshark.org/review/11117
Petri-Dish: Pascal Quantin <pascal.quantin@gmail.com>
Tested-by: Petri Dish Buildbot <buildbot-no-reply@wireshark.org>
Reviewed-by: Pascal Quantin <pascal.quantin@gmail.com>
(cherry picked from commit 2ae329a47b7f0ac94089c23e79c6b8bc18ba80ea)
Conflicts:
	epan/dissectors/packet-umts_fp.c
Reviewed-on: https://code.wireshark.org/review/11123
(cherry picked from commit 66f27abfa01ee0e252a5a4e0ffbff2e006c4943e)
Reviewed-on: https://code.wireshark.org/review/13749
Reviewed-by: Balint Reczey <balint@balintreczey.hu>
---
 epan/dissectors/packet-umts_fp.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/epan/dissectors/packet-umts_fp.c b/epan/dissectors/packet-umts_fp.c
index c622479..51b8fbe 100644
--- a/epan/dissectors/packet-umts_fp.c
+++ b/epan/dissectors/packet-umts_fp.c
@@ -3166,7 +3166,7 @@ dissect_hsdsch_channel_info(tvbuff_t *tvb, packet_info *pinfo, proto_tree *tree,
 
         /************************/
         /*Configure the pdus*/
-        for(i=0;i<number_of_pdus; i++){
+        for(i=0;i<number_of_pdus && i<MIN(MAX_MAC_FRAMES, MAX_RLC_CHANS); i++){
             macinf->content[i] = hsdsch_macdflow_id_mac_content_map[p_fp_info->hsdsch_macflowd_id]; /*MAC_CONTENT_PS_DTCH;*/
             macinf->lchid[i] = fake_lchid_macd_flow[p_fp_info->hsdsch_macflowd_id];/*Faked logical channel id 255 used as a mark it doesnt exists...*/
             macinf->fake_chid[i] = TRUE;    /**/
-- 
2.1.4

